src/components/Settings/Notifications/NotificationsWebhook/index.tsx

```diff
@@ -60,14 +59,26 @@ const NotificationsWebhook: React.FC = () => {
 
   const NotificationsWebhookSchema = Yup.object().shape({
     webhookUrl: Yup.string()
-      .required(intl.formatMessage(messages.validationWebhookUrl))
+      .when('enabled', {
+        is: true,
+        then: Yup.string()
+          .nullable()
+          .required(intl.formatMessage(messages.validationWebhookUrl)),
+        otherwise: Yup.string().nullable(),
+      })
```

Comment:
You made this sort of change in many places, but I am unsure how I feel about it UX-wise.

Users could actually save invalid data, but when they went to enable it, suddenly errors would spring up, telling them what they entered was wrong. But it just let them save it before? I think the validation should always be on if the inputs are visible. If we want to "remove" user input when the notification agent isn't enabled, it would be better to have something that hides all the inputs unless the type is actually enabled (which would make this toggled validation make sense)—something to think about for another PR.

Reply:
So this doesn't actually allow them to save invalid data. Rather, it just makes them optional (doesn't apply `required()`).  If a user enters a value in the field, it still must pass the valdation after/outside the `when()`.